### PR TITLE
Fill remaining missing hints & disallow partial hints

### PR DESCRIPTION
**Pull request by intgr**
Disallow untyped/incomplete function definitions (`disallow_untyped_defs = true`, `disallow_incomplete_defs = true`)

But using `_typeshed.Incomplete` instead of `Any` as the placeholder type.

## Related issues

* https://github.com/typeddjango/django-stubs/pull/1206
* Depends on #479

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -27,3 +26,19 @@ django_settings_module = scripts.drf_tests_settings
 
 [mypy-uritemplate.*]
 warn_unreachable = false
+
+[mypy-yaml.*]
+disallow_untyped_defs = false
+disallow_incomplete_defs = false
```
Comment (intgr) [inline mypy.ini:32]: I think these are necessary due to the way `pytest-mypy-plugins` works.

Mypy ordinarily ignores errors from site-packages imports, but they aren't ignored in tests.

**Review by sobolevn — APPROVED** [review]
Thanks! I skimmed through, this is not a detailed review :)
We can fix any annotation later if needed.

File: mypy.ini (hunk 1 above)
Comment (sobolevn) [reply mypy.ini:32]: This can be a new bug / feature request in `pytest-mypy-plugins`. Users should not see this.

Comment (intgr) [reply mypy.ini:32]: 👍 Reported issue https://github.com/typeddjango/pytest-mypy-plugins/issues/134

**Pull request merged**